src/rpc/blockchain.cpp

```diff
@@ -1032,128 +1032,250 @@ UniValue reconsiderblock(const UniValue& params, bool fHelp)
     sc.push_back(Pair("state", CSidechain::stateToString(scState)));
     sc.push_back(Pair("ceasing height", info.GetCeasingHeight()));
-    sc.push_back(Pair("creating tx hash", info.creationTxHash.GetHex()));
-    sc.push_back(Pair("created in block", info.creationBlockHash.ToString()));
+
+    if (bVerbose)
+    {
+        sc.push_back(Pair("creating tx hash", info.creationTxHash.GetHex()));
+        sc.push_back(Pair("created in block", info.creationBlockHash.ToString()));
+    }
+
     sc.push_back(Pair("created at block height", info.creationBlockHeight));
     sc.push_back(Pair("last certificate epoch", info.lastEpochReferencedByCertificate));
-    sc.push_back(Pair("last certificate hash", info.lastCertificateHash.GetHex()));
+
+    if (bVerbose)
+    {
+        sc.push_back(Pair("last certificate hash", info.lastCertificateHash.GetHex()));
+    }
+
     // creation parameters
     sc.push_back(Pair("withdrawalEpochLength", info.creationData.withdrawalEpochLength));
-    sc.push_back(Pair("wCertVk", HexStr(info.creationData.wCertVk)));
-    sc.push_back(Pair("customData", HexStr(info.creationData.customData)));
-    sc.push_back(Pair("constant", HexStr(info.creationData.constant)));
 
-    UniValue ia(UniValue::VARR);
-    BOOST_FOREACH(const auto& entry, info.mImmatureAmounts)
+    if (bVerbose)
     {
-        UniValue o(UniValue::VOBJ);
-        o.push_back(Pair("maturityHeight", entry.first));
-        o.push_back(Pair("amount", ValueFromAmount(entry.second)));
-        ia.push_back(o);
+        sc.push_back(Pair("wCertVk", HexStr(info.creationData.wCertVk)));
+        sc.push_back(Pair("customData", HexStr(info.creationData.customData)));
+        sc.push_back(Pair("constant", HexStr(info.creationData.constant)));
+
+        UniValue ia(UniValue::VARR);
+        for(const auto& entry: info.mImmatureAmounts)
+        {
+            UniValue o(UniValue::VOBJ);
+            o.push_back(Pair("maturityHeight", entry.first));
+            o.push_back(Pair("amount", ValueFromAmount(entry.second)));
+            ia.push_back(o);
+        }
+        sc.push_back(Pair("immature amounts", ia));
     }
-    sc.push_back(Pair("immature amounts", ia));
+    return true;
 }
 
-bool AddScInfoToJSON(const uint256& scId, UniValue& sc)
+bool FillScRecord(const uint256& scId, UniValue& scRecord, bool bOnlyAliv, bool bVerbosee)
 {
     CSidechain scInfo;
     CCoinsViewCache scView(pcoinsTip);
     if (!scView.GetSidechain(scId, scInfo)) {
         LogPrint("sc", "scid[%s] not yet created\n", scId.ToString() );
-        return false;
+        throw JSONRPCError(RPC_INVALID_PARAMETER, string("scid not yet created: ") + scId.ToString());
     }
     CSidechain::State scState = scView.isCeasedAtHeight(scId, chainActive.Height() + 1);
 
-    AddScInfoToJSON(scId, scInfo, scState, sc);
-    return true;
+    return FillScRecordFromInfo(scId, scInfo, scState, scRecord, bOnlyAliv, bVerbosee);
```

Comment:
two typos here `bOnlyAliv` which should be `bOnlyAlive` and `bVerbosee` which should be `bVerbose`

Reply:
Done